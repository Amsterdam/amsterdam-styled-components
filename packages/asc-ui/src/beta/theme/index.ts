import * as ascBetaTheme from './default'

export { default as ThemeProvider } from './ThemeProvider'
export { ascBetaTheme }
