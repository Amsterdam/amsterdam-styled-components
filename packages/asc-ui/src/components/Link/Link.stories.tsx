import * as React from 'react'
import { storiesOf } from '@storybook/react'
import Link from './Link'

storiesOf('Atoms/Typography/Link', module).add('default style', () => (
  <>
    <Link href="#">I am a link!</Link>
  </>
))
