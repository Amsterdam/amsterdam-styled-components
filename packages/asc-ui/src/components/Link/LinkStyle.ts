import styled, { css } from '@datapunt/asc-core'
import { themeColor } from '../../utils'
import {
  svgFill,
  FocusStyle,
  getFocusStyle,
  themeSpacing,
} from '../../utils/themeUtils'
import Typography, { TypographyProps } from '../Typography'
import IconStyle from '../Icon/IconStyle'

export type LinkVariant = 'inline' | 'blank' | 'with-chevron'

export type Props = {
  variant?: LinkVariant
  /**
   * @deprecated
   */
  linkType?: LinkVariant
  /**
   * @deprecated use prop onDarkBackground instead
   */
  color?: string
  focusStyle?: FocusStyle
  onDarkBackground?: boolean
} & TypographyProps

export const BlankLinkStyleCSS = css`
  display: inline-block;
  text-decoration: none;
  color: inherit;
`
export const InlineLinkStyleCSS = css`
  display: inline-block;
  color: ${themeColor('primary')};

  &:hover {
    color: ${themeColor('secondary')};
  }
`
export const DefaultLinkStyleCSS = css<Props>`
  display: inline-flex;
  text-decoration: none;
  font-weight: 700;
  color: ${({ color: colorOverride, theme }) =>
    themeColor('tint', 'level7', colorOverride)({ theme })};

  ${IconStyle} {
    margin: ${themeSpacing(1, 1, 0, 0)};
    ${({ color: colorOverride, theme, onDarkBackground }) =>
      onDarkBackground
        ? svgFill('tint', 'level1')({ theme })
        : svgFill('tint', 'level7', colorOverride)({ theme })};
  }

  &:hover {
    text-decoration: underline;
    ${({ color: colorOverride, theme, onDarkBackground }) =>
      css`
        color: ${onDarkBackground
          ? themeColor('tint', 'level1')({ theme })
          : themeColor('secondary', 'main', colorOverride)({ theme })};
        ${IconStyle} {
          ${onDarkBackground
            ? svgFill('tint', 'level1')({ theme })
            : svgFill('secondary', 'main', colorOverride)({ theme })}
        }
      `}
  }
`

export const LinkContent = styled.span`
  width: 100%; /* IE11 Fix */
`

export default styled(Typography)<Props>`
  ${({ focusStyle }) => getFocusStyle(focusStyle)}
  ${({ variant }) => {
    switch (variant) {
      case 'blank':
        return BlankLinkStyleCSS
      case 'inline':
        return InlineLinkStyleCSS
      default:
        return DefaultLinkStyleCSS
    }
  }}
`
