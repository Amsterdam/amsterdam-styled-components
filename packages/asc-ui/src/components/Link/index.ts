export { default as LinkStyle } from './LinkStyle'
export { default } from './Link'
