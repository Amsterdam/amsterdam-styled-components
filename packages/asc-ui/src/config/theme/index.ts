import { Theme } from '@amsterdam/asc-core'
import colors from './colors'
import typography from './typography'

export default Theme.getTheme(
  colors,
  typography,
)
