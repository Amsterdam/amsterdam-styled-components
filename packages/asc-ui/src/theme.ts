export default interface IThemeInterface {
  primaryColor: string
  secondaryColor: string
}
