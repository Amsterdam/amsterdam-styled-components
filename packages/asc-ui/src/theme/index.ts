import * as ascDefaultTheme from './default'

export { default as CompactThemeProvider } from './CompactThemeProvider'
export { default as ThemeProvider } from './ThemeProvider'
export { ascDefaultTheme }
