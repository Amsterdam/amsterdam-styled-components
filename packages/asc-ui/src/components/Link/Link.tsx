import { styledComponents } from '@datapunt/asc-core'
import React from 'react'
import LinkStyle from './LinkStyle'

export type Props = styledComponents.StyledProps<any> &
  React.HTMLProps<HTMLAnchorElement>

const Link: React.FC<Props> = ({ children, className, ...otherProps }) => (
  <LinkStyle as="a" className={className} {...otherProps}>
    {children}
  </LinkStyle>
)

export default Link
