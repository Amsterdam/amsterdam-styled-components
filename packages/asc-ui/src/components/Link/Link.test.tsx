import { render } from '@testing-library/react'
import Link from './Link'
import { ascDefaultTheme, themeColor } from '../../index'
import { ThemeProvider } from '../../theme'

describe('Link', () => {
  it('should render without any icons', () => {
    const { container } = render(<Link href="/">default link</Link>)

    expect(container.firstChild).toBeDefined()
    expect(container.querySelector('svg')).toBeNull()
  })

  it('should render an external or download icon', () => {
    const { container, rerender } = render(
      <Link icon="external" href="/">
        default link
      </Link>,
    )
    expect(container.querySelector('svg')).toBeDefined()

    rerender(
      <Link icon="download" href="/">
        default link
      </Link>,
    )
    expect(container.querySelector('svg')).toBeDefined()
  })

  it('should render the inline variant', () => {
    const { container } = render(
      <ThemeProvider>
        <Link variant="inline" href="/">
          default link
        </Link>
      </ThemeProvider>,
    )

    expect(container.firstChild).toHaveStyleRule(
      'color',
      themeColor('primary')({ theme: ascDefaultTheme }),
    )
  })

  it('should render the blank variant', () => {
    const { container } = render(
      <Link variant="blank" href="/">
        default link
      </Link>,
    )

    expect(container.firstChild).toHaveStyleRule('display', 'inline-block')
  })

  it('should render the an icon when passing the inList prop', () => {
    const { container } = render(
      <Link inList href="/">
        default link
      </Link>,
    )

    expect(container.querySelector('svg')).toBeDefined()
  })
})
