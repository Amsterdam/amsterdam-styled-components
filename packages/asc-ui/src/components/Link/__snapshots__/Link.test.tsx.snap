// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link should render 1`] = `
.c0 {
  margin-top: 0;
  margin-right: 0;
  margin-bottom: 0;
  margin-left: 0;
  color: #000000;
  font-weight: inherit;
  font-size: 1rem;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.25;
  margin-bottom: 0;
  font-stretch: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  color: #004699;
  display: inline-block;
}

.c0:focus {
  background-color: #FEC813;
}

.c0:hover {
  color: #ec0000;
}

<a
  class="c0"
  href="#"
>
  Foo
</a>
`;
